spec/sailfish_iptables_dbus_interface_description.xml

```diff
@@ -0,0 +1,464 @@
+<arg name="table" type="s" direction="in"/>
+<arg name="result" type="q" direction="out"/>
+<arg name="string" type="s" direction="out"/>
+</method>
+
+<!--
+Change policy of filter table INPUT chain. Parameter is the same as iptables
+command (ACCEPT/DROP)
+Sends signal PolicyChanged
+-->
+<method name="ChangeInputPolicy">
+<arg name="policy" type="s" direction="in"/>
+<arg name="result" type="q" direction="out"/>
+<arg name="string" type="s" direction="out"/>
+</method>
+
+<!--
+Change policy of filter table OUTPUT chain. Parameter is the same as iptables
+command (ACCEPT/DROP)
+Sends signal PolicyChanged
+-->
+<method name="ChangeOutputPolicy">
+<arg name="policy" type="s" direction="in"/>
+<arg name="result" type="q" direction="out"/>
+<arg name="string" type="s" direction="out"/>
+</method>
+
+<!--
+Following methods take following parameters as input:
+ NAME		TYPE	DESCRIPTION
+ ip		s		IP address or network with CIDR or dot notation (1.2.3.4/24
+ 					or 1.2.3.4/255.255.255.0)
+ port		q		Port number as unsigned 16bit integer
+ port		s		Port as string, single port or port range, separator is ":"
+ service	s		Service name, case insensitive, 
+ operation	s 		Perform operation add/remove, case insensitive, defaults to
+ 					add if empty string. Add appends the rule to iptables.
+ 					Remove removes the rule from iptables.
+ protocol	s 		tcp/udp, case insensitive (sctp not supported by sailfishos
+ 					iptables yet), usually defaults to TCP as tcp protocol is
+ 					defined first in /etc/services. Can be empty if service/
+ 					port is found from /etc/services.
+
+The methods are named with following convention:
+	Permission|Direction|Parameters
+	
+Permission:
+	Allow			Add rule to iptables with "-j ACCEPT"
+	Deny			Add rule to iptables with "-j DROP"
+	
+Direction:
+	Incoming		Add rule to chain INPUT of filter table
+	Outgoing		Add rule to chain OUTPUT of filter table
+	
+Parameters:
+	Ip			IP address as input
+	With			Combine two parameters
+	Port			Port parameter (integer q)
+	PortRange		Port or port range as parameter (string s)
+	Service			Service name as parameter
+	
+Each method sends signal RuleChanged if the rule was added to iptables.
+-->
+
+<!-- Add IP address or network to filter table INPUT chain as ACCEPT. -->
+<method name="AllowIncomingIp">
+<arg name="ip" type="s" direction="in"/>
+<arg name="operation" type="s" direction="in"/>
+<arg name="result" type="q" direction="out"/>
+<arg name="string" type="s" direction="out"/>
+</method>
+
+<!-- Add IP address or network with port to filter table INPUT chain as ACCEPT.
+Parameter protocol can be omitted if port is of some known service (/etc/services)
+-->
+<method name="AllowIncomingIpWithPort">
+<arg name="ip" type="s" direction="in"/>
+<arg name="port" type="q" direction="in"/>
+<arg name="protocol" type="s" direction="in"/>
+<arg name="operation" type="s" direction="in"/>
+<arg name="result" type="q" direction="out"/>
+<arg name="string" type="s" direction="out"/>
+</method>
```

Comment:
This is the same as `AllowIncomingIpWithPortRange` with the port range consisting of a single port, right? So it's unnecessary too.

Reply:
See prev. comment.